epoch1d/src/physics_packages/bremsstrahlung.F90

```diff
@@ -673,42 +676,103 @@ SUBROUTINE generate_photon(electron, z, iphoton)
         brem_array(brem_index)%e_table, brem_array(brem_index)%k_table, &
         brem_array(brem_index)%cdf_table, brem_array(brem_index)%state)
 
-    ! Calculate electron recoil
+    ! Is this photon to be added into the simulation?
+    add_photon = (photon_energy > photon_energy_min_bremsstrahlung &
+        .AND. produce_bremsstrahlung_photons)
+
+    ! Ensure photon_energy is a number we can handle at our precision
+    IF (photon_energy < c_tiny) photon_energy = c_tiny
+
+    ! Temporarily store photon momentum in the "new_photon" particle
+    CALL create_particle(new_photon)
+    photon_p = photon_energy / c
+    new_photon%part_p(1) = dir_x * photon_p
+    new_photon%part_p(2) = dir_y * photon_p
+    new_photon%part_p(3) = dir_z * photon_p
+
+    ! Rotate photon according to bremsstrahlung emission angular distribution
+    ! Only consider trajectory changes from e- with high enough energy to be
+    ! added into the simulation
+    IF (use_brem_scatter .AND. add_photon) THEN
```

Comment:
@Status-Mirror To restart the discussion from the old GitLab MR, could this be switched to:

```
IF (use_brem_scatter .AND. photon_energy > photon_energy_min_bremsstrahlung)
  scatter_theta = calc_scatter_theta(part_e)
  ! etc
```

I think that would allow users to include the scatter (on the electrons) without including the actual photons, and the second half of the if statement would avoid any concerns about getting stuck trying to sample a value for theta for (very) low energy photons.

Reply:
Yes this should be fine - the add_photon condition already required 
`photon_energy > photon_energy_min_bremsstrahlung`
so this won't make the sampling routines behave any differently. I just didn't realise people would want to model bremsstrahlung scattering without looking at the photons!

P.S. I've also noticed I forgot to update the comment on the "generate_photon" subroutine - it still says photons are emitted in the direction of electron motion. This could be changed to "generates a photon at the electron position, calculating the emission angle relative to electron momentum and the electron recoil if appropriate"

Reply:
I've made the changes to the logic for applying the scatter. Are you OK to update the comment you mention? Suspect you will do a more accurate job than me. Once that's done I'll do a review of the code and hopefully merge into 4.18-devel.